src/main/java/com/minecolonies/core/colony/Colony.java

```diff
@@ -1946,42 +1943,25 @@ public void setNameStyle(final String style)
     {
         this.nameStyle = style;
         this.markDirty();
-    }
-
-    /**
-     * Check if a civilian has a mismatched name pack.
-     *
-     * @param civilian the civilian to check.
-     * @param currentNamePack the current colony name pack.
-     * @return true if the name should be regenerated.
-     */
-    private boolean hasMismatchedNamePack(final ICivilianData civilian, final String currentNamePack)
-    {
-        if (civilian.hasSpecialName())
-        {
-            return false;
-        }
-        return civilian.getAssignedNamePack().isEmpty() || !civilian.getAssignedNamePack().equals(currentNamePack);
+        IMinecoloniesAPI.getInstance().getEventBus().post(new ColonyNameStyleChangedModEvent(this));
     }
 
     @Override
     public boolean hasCitizensWithMismatchedNamePack()
     {
-        final String currentNamePack = this.getNameStyle();
-        return citizenManager.getCitizens().stream()
-            .anyMatch(civilian -> hasMismatchedNamePack(civilian, currentNamePack));
+        return citizenManager.hasCitizensWithMismatchedNamePack();
```

Comment:
My original comments didn't mean to say just pass through the methods. No. Remove this entirely from the Colony, only the citizen manager should be allowed to know about the existence of these methods. People can fetch the citizen manager externally and shouldn't need passthrough via the colony.